test/integration/thesis_test.rb

```diff
@@ -167,6 +167,57 @@ def teardown
     assert_equal new_thesis_path, path
   end
 
+  test 'a confirmation email is sent on form submit' do
+    user = users(:basic)
+    thesis = theses(:two)
+
+    mock_auth(user)
+
+    ClimateControl.modify DISABLE_ALL_EMAIL: 'false' do
+      email = ReceiptMailer.receipt_email(thesis, user)
```

Comment:
I may be missing something about the testing environment - does calling this method - which itself calls `mail()` inside it, result in an email being sent due to this line?

Reply:
Hm... I actually don't think you're missing anything. This seems to be a case of pernicious copypasta from the receipt mailer test. It looks like the test is passing because it explicitly calls `receipt_email`, not because it does so while updating the thesis record. So, basically it's just doing the same thing as the receipt mailer test.

All that to say that this is broken. Thanks for the callout!

Reply:
Oh, never mind. On closer look, it _does_ still send the email without that line, so this is less broken than I thought.